src/components/Table/GridTable.tsx

```diff
@@ -270,141 +259,73 @@ export function GridTable<R extends Kinded, X extends Only<GridTableXss, X> = an
   const expandedColumnIds: string[] = useComputed(() => tableState.expandedColumnIds, [tableState]);
   const columnSizes = useSetupColumnSizes(style, columns, resizeTarget ?? resizeRef, expandedColumnIds);
 
-  // Make a single copy of our current collapsed state, so we'll have a single observer.
-  const collapsedIds = useComputed(() => tableState.collapsedIds, [tableState]);
```

Comment:
Collapsed-ness is handling within the `RowState.selfAndMaybeChildren` that flattens a row + its children (unless collapased).

The `selfAndMaybeChildren` also does the flattening that we used to do via visits/recursion.